hip_data_tools/apache/cassandra.py

```diff
@@ -32,6 +38,97 @@ def get_ssl_context(cert_path: str) -> SSLContext:
     return ssl_context
 
 
+_PYTHON_TO_CASSANDRA_DATA_TYPE_MAP = {
+    "Timestamp": "timestamp",
+    "str": "varchar",
+    "int64": "bigint",
+    "int32": "int",
+    "dict": "map",
+    "float64": "double",
+    "UUID": "UUID",
+}
+
+
+def _get_data_frame_column_types(data_frame):
+    data_frame_col_dict = {}
+    for col in data_frame:
+        data_frame_col_dict[col] = type(data_frame[col][0]).__name__
+    return data_frame_col_dict
+
+
+def convert_dataframe_columns_to_cassandra(data_frame):
+    """
+    Extracts a dictionary of column names and their cassandra data types from the dataframe
+    Args:
+        data_frame (DataFrame): the dataframe whose columns need to be extracted
+    Returns: dict
+    """
+    column_dtype = _get_data_frame_column_types(data_frame)
+    cassandra_columns = {key: _PYTHON_TO_CASSANDRA_DATA_TYPE_MAP[value] for (key, value) in
+                         column_dtype.items()}
+    return cassandra_columns
+
+
+def _pandas_factory(colnames, rows):
+    return pd.DataFrame(rows, columns=colnames)
```

Comment:
why use this? It's just a one line recreation of this single method?

Reply:
this callable needs to be passed in as a factory in the  Cassandra python API with that particular signature to convert the results into data frame. see usage.

Reply:
can you add a comment explaining this?